### PR TITLE
Always publish in directories instead of publishing individual files

### PR DESCRIPTION
**Pull request by lbussell**
Test run: https://dev.azure.com/dnceng/internal/_build/results?buildId=2412507&view=results [internal link]
Fixes https://github.com/dotnet/docker-tools/issues/1216

### CODE REVIEW

File: eng/common/templates/jobs/post-build.yml [hunk 1]
```diff
@@ -64,7 +64,7 @@ jobs:
       publicProjectName: ${{ parameters.publicProjectName }}
   - template: /eng/common/templates/steps/publish-artifact.yml@self
     parameters:
-      path: $(Build.ArtifactStagingDirectory)$(imageInfosSubDir)/image-info.json
+      path: $(Build.ArtifactStagingDirectory)$(imageInfosSubDir)
```
Comment (mthalman) [inline eng/common/templates/jobs/post-build.yml:67]: There are other files in this directory that we don't want included. Those are the image-info files from each job. We only want to publish the single combined image-info file here.

File: eng/common/templates/jobs/publish.yml [hunk 2]
```diff
@@ -67,25 +73,25 @@ jobs:
...
-      $(imageBuilder.pathArgs)
-      $(imageBuilder.commonCmdArgs)
+        '$(imageInfoContainerDir)/image-info.json'
+        --repo-prefix '$(publishRepoPrefix)'
+        --registry-creds '$(acr.server)=$(acr.userName);$(acr.password)'
+        --os-type '*'
+        --architecture '*'
+        $(dryRunArg)
+        $(imageBuilder.pathArgs)
+        $(imageBuilder.commonCmdArgs)
```
Comment (mthalman) [inline eng/common/templates/jobs/publish.yml:90]: I think this indentation might have messed up the command from running.

**Comment by lbussell** [comment]
New test run: https://dev.azure.com/dnceng/internal/_build/results?buildId=2413947&view=results

**Comment by lbussell** [comment]
This is ready for review again. [This is the most up-to-date test run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2414092&view=results). The warning shown about SBOMs is unrelated to the changes, not in our control. I ran [an additional short test run ](https://dev.azure.com/dnceng/internal/_build/results?buildId=2414441&view=results)to make sure that the warning went away.

**Review by mthalman — APPROVED** [review]

**Pull request merged**